Can't build arrayvec 0.7.5 - unresolved import `std::path` (for no std)

error[E0432]: unresolved import `std::path`
 --> C:\Users\evex\.cargo\registry\src\index.crates.io-6f17d22bba15001f\arrayvec-0.7.5\src\array_string.rs:8:10
  |
8 | use std::path::Path;
  |          ^^^^ could not find `path` in `std`

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -44,15 +44,40 @@ jobs:
         run: |
           cargo update -p serde_test --precise 1.0.163
           cargo update -p serde --precise 1.0.69
+      - name: Build
+        run: |
+          cargo build -v --no-default-features
+          cargo build --verbose --features "${{ matrix.features }}"
       - name: Tests
         run: |
-          cargo build --verbose --features "${{ matrix.features }}"
           cargo doc --verbose --features "${{ matrix.features }}" --no-deps
           cargo test --verbose --features "${{ matrix.features }}"
           cargo test --release --verbose --features "${{ matrix.features }}"
       - name: Test run benchmarks
         if: matrix.bench != ''
         run: cargo test -v --benches
+
+  nostd:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    strategy:
+      matrix:
+        include:
+          - rust: stable
+            target: thumbv6m-none-eabi
+            features: zeroize
+
+    name: nostd/${{ matrix.target }}/${{ matrix.rust }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ matrix.rust }}
+          targets: ${{ matrix.target }}
+      - name: Tests
+        run: |
+          cargo rustc "--target=${{ matrix.target }}" --no-default-features --features "${{ matrix.features }}"
+
 
   miri:
     runs-on: ubuntu-latest

--- a/src/array_string.rs
+++ b/src/array_string.rs
@@ -5,6 +5,7 @@ use std::fmt;
 use std::hash::{Hash, Hasher};
 use std::mem::MaybeUninit;
 use std::ops::{Deref, DerefMut};
+#[cfg(feature="std")]
 use std::path::Path;
 use std::ptr;
 use std::slice;
@@ -502,6 +503,7 @@ impl<const CAP: usize> fmt::Debug for ArrayString<CAP>
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { (**self).fmt(f) }
 }
 
+#[cfg(feature="std")]
 impl<const CAP: usize> AsRef<Path> for ArrayString<CAP> {
     fn as_ref(&self) -> &Path {
         self.as_str().as_ref()

